### PR TITLE
Huffman Algorithm

### LINKED ISSUE #7000: Huffman algorithm
## 💥 Proposal

(A clear and concise description of what the proposal is.)
I would like to add the code for huffman algorithm in C, under GSSOC'21
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

(Write your answer here.)
yes

### PR DESCRIPTION
**Pull request by Shatakshijain**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
yes
(Write your answer here.)

### Description

(Write your answer here.)
I have added the code for huffman algorithm in C.
### Checklist

- [ ] I've read the contribution guidelines.
- [ ] I've checked the issue list before deciding what to submit.
- [ ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
Closes: #7000 
(Write your answer here.)

### CODE REVIEW

File: C/ds/huffman.c [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+  int i;
+  for (i = 0; i < n; ++i)
+    printf("%d", arr[i]);
+
+  printf("\n");
+}
+
+int main() {
+  char arr[] = {'A', 'B', 'C', 'D'};
+  int freq[] = {5, 1, 6, 3};
```
Comment (siddharth25pandey) [inline C/ds/huffman.c]: Use Dynamic Input

File: C/ds/huffmanalgo.c [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+struct Heap {
+  long long size;
+  long long capacity;
+  struct Node **array;
+};
+
+// Create nodes
+struct Node *newNode(char item, long long freq) {
+  struct Node *temp = (struct Node *)malloc(sizeof(struct Node));
```
Comment (siddharth25pandey) [inline C/ds/huffmanalgo.c:21]: Remove the Duplicate File

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@Shatakshijain  Do the suggested changes and also add the Update in README file

**Comment by siddharth25pandey** [comment]
> @Shatakshijain Do the suggested changes and also add the Update in README file

@Shatakshijain Not done yet

**Review by ankitaggarwal23 — COMMENTED** [review]
Thank you for your valuable time.😊

**Pull request closed without merging**